### PR TITLE
MAINT: Remove unused imports and unreachable code

### PR DESCRIPTION
**Pull request by donno2048**
List of the minor fixes by order of commit:

- Remove unnecessary imports
- Remove unnecessary variables
- Fix usage of `keys` and `elif`
- Reimport `_set_madvise_hugepage`

### CODE REVIEW

**Review by donno2048 — COMMENTED** [review]
The second commit message was meant to be `Remove unnecessary variables`

**Comment by bashtage** [comment]
The entire py3k file should be removed.

File: numpy/f2py/crackfortran.py [hunk 1]
```diff
@@ -1540,8 +1539,6 @@ def markinnerspaces(line):
             cc = c
         if c == cc:
             f = f + 1
-        elif c == cc:
```
Comment (bashtage) [inline numpy/f2py/crackfortran.py:1543]: This looks like a bug and probably should be simply deleted.

Comment (eric-wieser) [reply numpy/f2py/crackfortran.py:1543]: I'm wary of changing this code without understanding what was actually intended here. Obviously the current code is nonsense, but cleaning up nonsense while preserving behavior makes it harder to work out whether that behavior is what the nonsense was intended to do in the first place.

Does the git blame reveal the commit that introduced this duplication?

**Comment by eric-wieser** [comment]
> The entire py3k file should be removed.

Doing that will break any downstream packages that import `numpy.compat` and use these constants - unless you mean "this PR should not touch `py3k.py`", which I agree with.

**Comment by bashtage** [comment]
> Doing that will break any downstream packages that import `numpy.compat` and use these constants - unless you mean "this PR should not touch `py3k.py`", which I agree with.

I though compat and compat.py3k were considered private?

File: numpy/f2py/crackfortran.py (hunk 1 above)
Comment (bashtage) [reply numpy/f2py/crackfortran.py:1543]: > I'm wary of changing this code without understanding what was actually intended here. Obviously the current code is nonsense, but cleaning up nonsense while preserving behavior makes it harder to work out whether that behavior is what the nonsense was intended to do in the first place.
> 
> Does the git blame reveal the commit that introduced this duplication?

I fully agree.  I somehow short circuited the **not** in my op.  It definitely should be investigated.

**Comment by bashtage** [comment]
@donno2048 The PR title should be more meaningful and use one of the leading descriptors, such as `MAINT: Remove unused imports and unreachable code`.

**Comment by bashtage** [comment]
This reason for thinking these modules are private

https://github.com/numpy/numpy/blob/623bc1fae1d47df24e7f1e29321d0c0ba2771ce0/numpy/tests/test_public_api.py#L185-L188

**Comment by eric-wieser** [comment]
From the 1.19 release notes:

> The shims in numpy.compat will remain to support third-party packages, but they may be deprecated in a future release.

So we could probably pull the trigger on deprecating these, but I don't think we should remove them immediately, and certainly not as part of a misc cleanup PR like this one.

**Comment by donno2048** [comment]
> The PR title should be more meaningful and use one of the leading descriptors, such as `MAINT: Remove unused imports and unreachable code`.

Makes sense...

**Comment by rgommers** [comment]
> I though compat and compat.py3k were considered private?

Yep, but unfortunately no one cares. Even SciPy uses `numpy.compat` in multiple places.

**Comment by mattip** [comment]
@donno2048 please relate to the comments on this PR in order to move it forward.

**Comment by donno2048** [comment]
> @donno2048 please relate to the comments on this PR in order to move it forward.

The only thing not resolved is the removal of `elif c == cc` at _numpy/f2py/crackfortran.py_

**Comment by mattip** [comment]
Exactly. The next round of review can start when you revert that removal or dig into git history, discover where that change was made, and provide a reasonable analysis of what could have caused this to appear.

**Comment by donno2048** [comment]
This `if`-`elif` condition is there from the first creation of the file https://github.com/numpy/numpy/commit/571c53f4101a82805bcbe0b8378dd6b147a5d797

**Comment by mattip** [comment]
Ahh, I see [this line](https://github.com/numpy/numpy/commit/571c53f4101a82805bcbe0b8378dd6b147a5d797#diff-600972d74f0d6654cca3a432eaf2620e9bd602e6db3c71dd44a0399d76c52db6R1182) is the equivalent one. @pearu any thoughts?

**Comment by pearu** [comment]
The code
```
        if c == cc:
            f = f + 1
        elif c == cc:
            f = f - 1
```
should (most likely) read:
```
        if c == cc:
            f = not f
```

The function `markinnerspaces` documentation is
```
Return line with all spaces of quoted substrings replaced with `@_@`.

Examples::

  markinnerspaces("a b") -> "a b"
  markinnerspaces("a b 'c d' e") -> "a b 'c@_@d' e"
  markinnerspaces("a b \'c d\' e") -> "a b \'c d\' e"
```

**Comment by pearu** [comment]
Notice that https://github.com/numpy/numpy/pull/15208 already fixes the `markinnerspaces` issue. So, it would make sense to land #15208 and undo related code changes here.

**Comment by donno2048** [comment]
> Notice that #15208 already fixes the `markinnerspaces` issue. So, it would make sense to land #15208 and undo related code changes here.

I have reversed this change

**Review by mattip — APPROVED** [review]
LGTM, pending CI passing

**Review by rossbar — APPROVED** [review]
Thanks @donno2048

**Pull request merged**